client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,181 @@
-	} );
-
-	test( 'should have .has-call-to-action class if callToAction is defined', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 1 );
-	} );
-
-	test( 'should not have .has-call-to-action class if callToAction is null', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ null } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 0 );
+		renderWithRedux( <Banner { ...props } dismissPreferenceName="banner-test" /> );
+		expect( screen.getByLabelText( 'Dismiss' ) ).toBeVisible();
 	} );
 
 	test( 'should render a <Button /> when callToAction is specified', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Buy something!' } /> );
-		expect( comp.find( Button ) ).toHaveLength( 1 );
+		render( <Banner { ...props } callToAction="Buy something!" /> );
+		expect( screen.getByRole( 'button' ) ).toHaveTextContent( 'Buy something!' );
 	} );
 
 	test( 'should not render a <Button /> when callToAction is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( Button ) ).toHaveLength( 0 );
+		render( <Banner { ...props } /> );
+		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
 	} );
 
 	test( 'should have .is-jetpack class and JetpackLogo if jetpack prop is defined', () => {
 		const { plan, ...propsWithoutPlan } = props;
-		const comp = shallow( <Banner { ...propsWithoutPlan } jetpack /> );
-		expect( comp.find( '.is-jetpack' ) ).toHaveLength( 1 );
-		expect( comp.find( 'JetpackLogo' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...propsWithoutPlan } jetpack /> );
+		expect( container.firstChild ).toHaveClass( 'is-jetpack' );
+		expect( container.getElementsByClassName( 'jetpack-logo' ) ).toHaveLength( 1 );
 	} );
 
 	test( 'should render have .is-horizontal class if horizontal prop is defined', () => {
-		const comp = shallow( <Banner { ...props } horizontal /> );
-		expect( comp.find( '.is-horizontal' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } horizontal /> );
+		expect( container.firstChild ).toHaveClass( 'is-horizontal' );
 	} );
 
 	test( 'should render a <PlanPrice /> when price is specified', () => {
-		const comp = shallow( <Banner { ...props } price={ 100 } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 1 );
+		render( <Banner { ...props } price={ 100 } /> );
+		expect( screen.queryByTestId( 'banner-price-plan-mock' ) ).toHaveTextContent( '100' );
 	} );
 
 	test( 'should render two <PlanPrice /> components when there are two prices', () => {
-		const comp = shallow( <Banner { ...props } price={ [ 100, 80 ] } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 2 );
+		render( <Banner { ...props } price={ [ 100, 80 ] } /> );
+		expect( screen.queryAllByTestId( 'banner-price-plan-mock' ) ).toHaveLength( 2 );
```

Comment:
@tyxla - not sure if I implemented the PlanPrice component mock correctly (which is why I added it as a separate commit), but if I did, I'm not confident checking the array length is the best approach for this test. I did this in the meantime to get a pass in order to receive feedback on the mock and the rest of the changes. 

But if the mock is correct, I thought it would make sense to still check for the two price values. I thought I could use the method [`expect.arrayContaining`](https://jestjs.io/docs/expect#expectarraycontainingarray) but didn't succeed. It could be due to the other methods I've tried instead of `toEqual` shown in the docs example. I tried `toContain` as well since the docs say:

> `.toContain` can also check whether a string is a substring of another string.

But I didn't have any luck with that either. This is what is received when querying for the test id: 

```   
Expected value: ArrayContaining ["100", " 80"]
Received array: [<div data-testid="banner-price-plan-mock">100</div>, <div data-testid="banner-price-plan-mock">80</div>]
```

Reply:
I think you're in the right direction. We can actually leave it as-is and it'll be a literal translation of the original test, which is just fine for the migration purposes.

If you'd like to get it a bit further, there is just a bit more we can tweak to actually thoroughly test the props we pass in this branch of the logic.

The thing is that in `<Banner />` we render 2 `<PlanPrice />` components, one with `original` and one with `discounted` props. We could take those into account (perhaps have them passed to the resulting classname in our mock) and then add additional test assertions to verify that we have one of each. That will ensure that we properly pass the expected props to `PlanPrice`. For another bonus point, you could do an assertion for the price of both, verifying that it's the expected one in each of the `PlanPrice` instances. 

`expect.arrayContaining` could work, but when it comes to testing React component instances, it's lacking, and it's difficult (and counterintuitive) to get the inner objects to match, so I wouldn't recommend it. I think what you did with `toHaveLength` makes sense. From then on, you can just pick the first element from the collection (as simple as `[ 0 ]`), and then use `toHaveTextContent()` on it to compare the price. Then you can pass the `original` and `discounted` to another attribute - the `class` for example. If you do that, you can use `toHaveAttribute( 'class', 'discounted' )` to further check if it renders as the original price or discounted price.

Let me know if that makes sense.

Reply:
That makes sense (well, we'll know if it did if the changes I implemented are correct 😅 )!